src/swaps/liqualityboost/LiqualityBoostSwapProvider.js

```diff
@@ -0,0 +1,290 @@
+    const swapOneInchFormat = { ...swap, from: swap.toNativeAsset, fromAmount: swap.toNativeAssetAmount, fromAccountId: swap.toAccountId, slippagePercentage }
+    switch (swap.status) {
+      case 'INITIATED':
+        updates = await this.liqualitySwapProvider.reportInitiation({ swap: swapLiqualityFormat, network, walletId })
+        break
+
+      case 'INITIATION_REPORTED':
+        updates = await withInterval(async () => this.liqualitySwapProvider.confirmInitiation({ swap: swapLiqualityFormat, network, walletId }))
+        break
+
+      case 'INITIATION_CONFIRMED':
+        updates = await withLock(store, { item: swap, network, walletId, asset: swap.from },
+          async () => this.liqualitySwapProvider.fundSwap({ swap: swapLiqualityFormat, network, walletId }))
+        break
+
+      case 'FUNDED':
+        updates = await withInterval(async () => this.liqualitySwapProvider.findCounterPartyInitiation({ swap: swapLiqualityFormat, network, walletId }))
+        break
+
+      case 'CONFIRM_COUNTER_PARTY_INITIATION':
+        updates = await withInterval(async () => this.liqualitySwapProvider.confirmCounterPartyInitiation({ swap: swapLiqualityFormat, network, walletId }))
+        break
+
+      case 'READY_TO_CLAIM':
+        updates = await withLock(store, { item: swap, network, walletId, asset: swap.to },
+          async () => this.liqualitySwapProvider.claimSwap({ swap: swapLiqualityFormat, network, walletId }))
+        break
+
+      case 'WAITING_FOR_CLAIM_CONFIRMATIONS':
+        updates = await withInterval(async () => this.finalizeLiqualitySwapAndStartOneinch({ swap: swapLiqualityFormat, network, walletId }))
+        break
+
+      case 'WAITING_FOR_REFUND':
+        updates = await withInterval(async () => this.liqualitySwapProvider.waitForRefund({ swap: swapLiqualityFormat, network, walletId }))
+        break
+
+      case 'GET_REFUND':
+        updates = await withLock(store, { item: swap, network, walletId, asset: swap.from },
+          async () => this.liqualitySwapProvider.refundSwap({ swap: swapLiqualityFormat, network, walletId }))
+        break
+
+      case 'WAITING_FOR_REFUND_CONFIRMATIONS':
+        updates = await withInterval(async () => this.liqualitySwapProvider.waitForRefundConfirmations({ swap: swapLiqualityFormat, network, walletId }))
+        break
+      case 'SEND_ONEINCH_SWAP':
+        updates = await withLock(store, { item: swap, network, walletId, asset: swap.from },
+          async () => this.oneinchSwapProvider.sendSwap({ quote: swapOneInchFormat, network, walletId }))
+        break
+      case 'WAITING_FOR_SWAP_CONFIRMATIONS':
+        updates = await withInterval(async () => this.oneinchSwapProvider.waitForSwapConfirmations({ swap: swapOneInchFormat, network, walletId }))
+        break
+    }
+
+    return updates
+  }
+
+  static txTypes = {
+    SWAP_INITIATION: 'SWAP_INITIATION',
```

Comment:
```
...LiqualitySwapProvider.txTypes,
...OneInchSwapProvider.txTypes
```
? 😂

Reply:
Yep, this can work. 

I was thinking that I will need more types, but then it end up in having the same as these in liqualitySwapProvider and OneInch swap provider